mlflow/utils/validation.py

```diff
@@ -3,26 +3,33 @@
 """
 import re
 
-# Regex for metric and parameter names: we only allow [a-zA-Z0-9_-] ([\w-]), / and . characters,
+# Regex for metric and parameter names: we only allow [a-zA-Z0-9_- ] ([\w-]), / and . characters,
 # and we do not allow adjacent, leading or trailing / or path components that are empty or only
-# contain dots. The goal is to make our names be valid, unique path names too in order to treat
-# them as such in the UI and file store.
-_METRIC_AND_PARAM_NAME_REGEX = re.compile(r"^([\w.-]*[\w-][\w.-]*)(/[\w.-]*[\w-][\w.-]*)*$")
+# contain dots or whitespace. The goal is to make our names be valid, unique path names too in
+# order to treat them as such in the UI and file store.
+_METRIC_AND_PARAM_NAME_REGEX = re.compile(
+    r"^([\w.\- ]*[\w\-][\w.\- ]*)(/[\w.\- ]*[\w\-][\w.\- ]*)*$")
 
 # Regex for valid run IDs: must be a 32-character hex string.
 _RUN_ID_REGEX = re.compile(r"^[0-9a-f]{32}$")
 
+# An error message which strives to be concise and mostly-correct.
+CONCISE_EXPLANATION_MESSAGE = (
+    "Names must contain at least one alphanumeric, underscore (_) or dash (-), and may"
+    " also include periods (.), whitespace ( ), and slashes (/)."
```

Comment:
Unfortunately this doesn't say why "//" or ".." would be bad. I would just say names can contain [list of characters here] and must be valid, unique path names if they contain slashes. Or we can raise a different error based on what the user gives: for example name can't contain // or .. or whatever.

Reply:
The one who tries to create a metric with name `//` deserves all the suffering they receive.

Reply:
(My aim was to clarify for most users what the constraints were -- can change them to be more specific as you suggested, it's just less concise, which is also fine.)

Reply:
I think we could also just keep a bigger version in the docs, and say "invalid metric name" here. Then people will look in the docs for the larger explanation. But this was one reason I suggested allowing spaces in each path component, to make the rule easier to explain (even if metrics named `  `  and `    ` will not be that useful).